### PR TITLE
feat: azure support

### PR DESCRIPTION
**Pull request by chrishumanitec**
Added handling of Azure Container Registry

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -82,6 +86,15 @@ jobs:
           gcloud auth configure-docker ${{ env.GCP_GAR_HOST }} --quiet
           echo "CONTAINER_REGISTRY=${{ env.GCP_GAR_NAME }}" >> "$GITHUB_ENV"
 
+      - if: ${{ vars.CLOUD_PROVIDER == 'azure' }}
+        name: login to azure acr
+        run: |
+          echo "${{ secrets.ACR_PUSH_SP_PASSWORD }}" | docker login \
+            ${{ secrets.ACR_SERVER_NAME }} \
+            -u ${{ secrets.ACR_PUSH_SP_ID }} \
+            --password-stdin
```
Comment (johanneswuerbach) [inline .github/workflows/deploy.yaml:95]: We generally try to use workload identity and avoid hardcoding passwords, it seems this is also supported on azure https://github.com/Azure/login?tab=readme-ov-file#login-with-openid-connect-oidc-recommended

Comment (chrishumanitec) [reply .github/workflows/deploy.yaml:95]: Okay. Then I would close this PR. Just enabling this for people to use Backstage with Azure.

Comment (johanneswuerbach) [reply .github/workflows/deploy.yaml:95]: Moving to draft for now.

Comment (mathieu-benoit) [reply .github/workflows/deploy.yaml:95]: Yup, I think we should start with this one, in order to have this officially supporting Azure, and then later figure out the passwordless way to do that.

**Comment by johanneswuerbach** [comment]
Replaced by https://github.com/humanitec-architecture/backstage/pull/15

**Pull request closed without merging**